src/js/components/Notification/Notification.js

```diff
@@ -45,39 +94,85 @@ const Notification = ({ message, onClose, id, status, title, toast }) => {
   const { color: closeIconColor } = theme.notification.close;
 
+  const { direction, truncate, separator } = toast
+    ? theme.notification.toast
+    : theme.notification;
+
+  const TextWrapper = direction === 'row' ? Text : Fragment;
+  const textWrapperProps = direction === 'row' ? { truncate } : {};
+
+  // notification is built with two child boxes that contain:
+  // 1. icon + text (wrapped in button when clickable)
+  // 2. close button
+  // pad needs to be applied to the child boxes to ensure pad is included
+  // in the clickable region, but we don't want to apply extra padding
+  // between the icon + text and the button.
+  const { pad } = theme.notification.container;
+  let textPad;
+  let closeButtonPad;
+  if (onClose) [textPad, closeButtonPad] = adaptThemeStyle(pad, theme);
+  else textPad = pad;
+
+  let message;
+  if (messageProp && !truncate && direction !== 'row')
+    message = (
+      <Paragraph {...theme.notification.message}>{messageProp}</Paragraph>
+    );
+  else if (messageProp)
+    message = (
+      <Text {...theme.notification.message} truncate={truncate}>
+        {messageProp}
+      </Text>
+    );
+
   let content = (
     <Box
       {...theme.notification.container}
       {...(toast ? { ...theme.notification.toast.container } : {})}
+      background={
+        !toast && background
+          ? background
+          : theme.notification.container.background
+      }
+      // let internal box control pad so clickable region includes pad
+      pad={undefined}
       direction="row"
+      gap="small"
     >
-      <Box {...theme.notification.iconContainer}>
-        <StatusIcon color={color} />
-      </Box>
-      <Box
-        {...theme.notification.textContainer}
-        align="start"
-        direction="row"
-        justify="between"
-        flex
-      >
-        <Box>
-          <Text {...theme.notification.title}>{title}</Text>
-          {message && (
-            <Paragraph {...theme.notification.message}>{message}</Paragraph>
-          )}
+      {/* separate from onClose button to avoid nested interactive elements */}
+      <IconTextContainer href={href} onClick={onClick}>
+        <Box direction="row" pad={textPad} flex>
+          <Box {...theme.notification.iconContainer}>
+            <StatusIcon color={color} />
+          </Box>
+          <Box {...theme.notification.textContainer}>
+            <Box>
+              <TextWrapper {...textWrapperProps}>
+                {title && <Text {...theme.notification.title}>{title}</Text>}
+                {message && title && direction === 'row' && separator && (
+                  <Text>{separator}</Text>
+                )}
+                {message}
+              </TextWrapper>
+            </Box>
+          </Box>
+        </Box>
+      </IconTextContainer>
+      {onClose && (
+        <Box pad={closeButtonPad}>
+          <Box {...theme.notification.textContainer}>
```

Comment:
It seems strange to have nested Boxes here, can they be merged?

Reply:
The only reason I hadn't was for backwards compatibility. The `pad` on the outer box is technically coming from `theme.notification.container` (being applied here since we now separate the icon+text and button). so presumably someone could've had something defined for both theme objects and we would be reducing it down to just one of the `pad`s.

Reply:
So the `textContainer` might also have a `pad`, and we need to account for both?

Reply:
Yes